### PR TITLE
refactor(wallet): Implement Rust Address Generator

### LINKED ISSUE #1617: Replace nodejs-mobile-react-native functions with Rust functions via UniFFI

### PR DESCRIPTION
**Pull request by coreyphillips**
### Description
- Removes `nodejs-mobile` dependency and related files.
- Adds `react-native-address-generator` and implements it accordingly.
- The `BitcoinActions` class was moved from `nodejs-assets/nodejs-project` to `utils/bitcoin-actions` and updated accordingly.
- This update brings with it a significant performance improvement when generating addresses.
- If ya'll agree that this way of generating addresses in Bitkit is an improvement over the previous `nodejs-mobile` implementation and approve the PR we'll just need to import/move the `react-native-address-generator` library to the `synonymdev` name and make it public to continue working on from there moving forward. 

### Linked Issues/Tasks
- #1617

### Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (improving code without creating new functionality)

### Tests
- [x] No test

### QA Notes
- Please ensure address generation still works as expected on iOS & Android for each address type and network.

### CODE REVIEW

**Review by limpbrains — APPROVED** [review]

File: __tests__/wallet-restore.ts
Comment (pwltr) [inline __tests__/wallet-restore.ts:1]: What's the reason to remove these tests?

File: src/utils/bitcoin-actions/index.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		path,
+		selectedNetwork = this.selectedNetwork,
+	}): Promise<Result<string>> {
+		try {
+			if (!mnemonic) {
+				return err('No mnemonic specified');
+			}
+			if (!selectedNetwork) {
+				return err('No network specified');
+			}
```
Comment (pwltr) [inline src/utils/bitcoin-actions/index.ts]: Could remove these checks for `selectedNetwork`. It is never undefined

File: src/utils/bitcoin-actions/index.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+			this.mnemonic = mnemonic;
+			this.selectedNetwork = selectedNetwork;
+			this.bip39Passphrase = bip39Passphrase;
+			return ok('Successfully setup bitcoin-actions.');
+		} catch (e) {
+			return err(e);
+		}
+	}
```
Comment (pwltr) [inline src/utils/bitcoin-actions/index.ts]: This method could be covered by the `constructor`:

```js
const bitcoinActions = new BitcoinActions({
  mnemonic,
  selectedNetwork,
  bip39Passphrase
});
```

**Review by pwltr — CHANGES_REQUESTED** [review]

File: __tests__/wallet-restore.ts
Comment (coreyphillips) [reply __tests__/wallet-restore.ts:1]: That logic is now handled in beignet and tested [there](https://github.com/synonymdev/beignet/tree/master/tests).

File: src/utils/bitcoin-actions/index.ts (hunk 1 above)
Comment (coreyphillips) [reply src/utils/bitcoin-actions/index.ts]: Updated [here](https://github.com/synonymdev/bitkit/pull/1717/commits/15e21387224e6bedb21211878a0e65c24c5db886).

File: src/utils/bitcoin-actions/index.ts (hunk 2 above)
Comment (coreyphillips) [reply src/utils/bitcoin-actions/index.ts]: Updated [here](https://github.com/synonymdev/bitkit/pull/1717/commits/15e21387224e6bedb21211878a0e65c24c5db886).

**Review by pwltr — APPROVED** [review]

**Pull request merged**